### ISSUE TITLE
Bug: 0 in control.bdg even when slocal and llocal are NOT OFF.

### ISSUE THREAD

**Issue by taoliu** [taoliu/MACS#303]
I'm having an issue that I think is related to this. There is a short 17bp region in the middle of chr7 in one of my control bdg files that has a value of 0. I am using a treatment bed file as input with no control bam. I did not turn off slocal or llocal and I did not set nolambda. How is there a 0 value in the control bdg? Shouldn't the genome background be substitute if there aren't any nearby reads in that region?

_Originally posted by @kwcurrin in https://github.com/taoliu/MACS/issues/150#issuecomment-520017299_

**Comment by taoliu** [comment]
@kwcurrin I created this new issue ticket from #150. Could you provide more information on that? Such as the version, and the chunk of control bdg and treat bdg around that 17bp region?

**Comment by kwcurrin** [comment]
Hello,

I think I figured out this issue. Sometimes the bedgraph files extend past chr edges, which is a known problem. I previously used slopBed to remove these. However, in this new sample, there was a rare case where an entire bedgraph record was outside of the chr edges, not just the end position. There was a bedtools issue that caused slop to behave incorrectly when a record was completely outside of chr edges. I was able to fix this by using slopBed, and then bedClip. The resulting file does not have 0s in the background. I think the 0s were being filled in by another bedtools tool, unionBedGraphs, after slopBed created incorrect records.

Sorry for the incorrect issue title!
